src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable camelcase */
+/* eslint-disable no-useless-catch */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+import axios from 'axios';
+
+const initialState = {
+  books: [],
+  isLoading: true,
+};
+const itemId = 'SERz0T6MOlfncfi0umcc';
+
+const url = `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${itemId}/books`;
+
+export const getBooksItems = createAsyncThunk(
+  'books/getBooksItems',
+  async () => {
+    try {
+      const res = await axios.get(`${url}`);
+      return res.data;
+    } catch (error) {
+      return error;
+    }
+  },
+);
+
+export const addBook = createAsyncThunk('books/addBook', async (book) => {
+  try {
+    const res = await axios.post(`${url}`, book, {
+      'Content-Type': 'application/json',
+    });
+    return res.data;
+  } catch (error) {
+    return error;
+  }
+});
+
+export const selectBooks = (state) => state.books.books;
+
+export const selectIsLoading = (state) => state.books.isLoading;
+
+export const removeBook = createAsyncThunk(
+  'books/removeBook',
+  async (item_id) => {
+    try {
+      await axios.delete(`${url}/${item_id}`);
+      return item_id;
+    } catch (error) {
+      return error;
+    }
+  },
+);
+
```

Comment:
- [ ] The book add and fetch are well implemented. Books can be stored on the API and reloading the browser fetches the book. But the delete functionality does not really work. Whenever I click the delete button for just one book, the page goes blank as if all the books are removed (kindly fix this also). But when I refresh the page all the books are retrieved and displayed. Kindly fix that.

Reply:
@DammyShittu Please I am still struggling to remove one specific item book, I tried with one of my standUP team but nothing. Can you give some hints or any resources if you have time?